webapp/channels/src/components/common/hooks/useOpenPricingModal.ts

```diff
@@ -2,24 +2,33 @@
 // See LICENSE.txt for license information.
 
 import {useCallback} from 'react';
-import {useDispatch, useSelector} from 'react-redux';
+import {useSelector} from 'react-redux';
 
 import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
 
 import {trackEvent} from 'actions/telemetry_actions';
-import {openModal} from 'actions/views/modals';
 
-import PricingModal from 'components/pricing_modal';
+import {TELEMETRY_CATEGORIES} from 'utils/constants';
 
-import {ModalIdentifiers, TELEMETRY_CATEGORIES} from 'utils/constants';
+import useCWSAvailabilityCheck, {CSWAvailabilityCheckTypes} from './useCWSAvailabilityCheck';
 
 export type TelemetryProps = {
     trackingLocation: string;
 }
 
-export default function useOpenPricingModal() {
-    const dispatch = useDispatch();
+export type UseOpenPricingModalReturn = {
+    openPricingModal: (telemetryProps?: TelemetryProps) => void;
+    isAirGapped: boolean;
+}
+
+export default function useOpenPricingModal(): UseOpenPricingModalReturn {
     const isCloud = useSelector(isCurrentLicenseCloud);
+    const cwsAvailability = useCWSAvailabilityCheck();
+
+    const isAirGapped = cwsAvailability === CSWAvailabilityCheckTypes.Unavailable;
```

Comment:
There are no edge cases we care about here where the CWS can't be reached for some super strange reason, but the rest of the internet can be? I can't think of anything really, but just wanted to check.

Reply:
Yes, CWS cannot be reached if in Team edition as this check relies on the enterprise interfaces. For team edition we do want to show link outs to Pricing page